File: 119-pascals-triangle-ii/pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>> pascalTriangle(rowIndex + 1);
        pascalTriangle[0].push_back(1);
        for (int currentRow = 1; currentRow <= rowIndex; currentRow++) {
            pascalTriangle[currentRow].push_back(1);
            vector<int>& currentRowList = pascalTriangle[currentRow];
            vector<int>& previousRowList = pascalTriangle[currentRow - 1];
            for (int j = 1; j < previousRowList.size(); j++) {
                int sum = previousRowList[j] + previousRowList[j - 1];
                currentRowList.push_back(sum);
            }

                currentRowList.push_back(1);
        }

        return pascalTriangle[rowIndex];
    }
};

File: 119-pascals-triangle-ii/pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        x = [1]
        y = 1
        for i in range(1, rowIndex+1):
            z = y*(rowIndex-i+1)//i
            x.append(z)
            y = z
        return x